src/web/app/pipes/search-terms-highlighter.pipe.ts

```diff
@@ -0,0 +1,66 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+    name: 'highlighter',
+})
+export class SearchTermsHighlighterPipe implements PipeTransform {
+
+    /**
+     * Transforms text to add highlighting styling.
+     *
+     * @param value text to be transformed if it contains search terms
+     * @param args search terms entered by user in the search bar
+     * @param partial optional argument, true when text should be highlighted on a partial match,
+     *                by default, text is only highlighted on a full word match
+     * @returns transformed text with styling added if search terms were found
+     */
+    transform(value: any, args: any, partial?: boolean): unknown {
```

Comment:
Should avoid `any` or `unknown` as much as possible if type is known, `value`, `args` and return type are of type  `string`?